linode/manager
# fix: [M3-8588] - Fix `value` selection issue in `ImageSelect`

## Description 📝
This PR fixes the bug in the `ImageSelect` component which resulted in the selected `value` getting cleared after the input lost focus.

## Changes  🔄
List any change relevant to the reviewer.
- Add logic to handle the case when `value` is `undefined` initially.
- Remove leftover `default export` declarations.
- Replace single letter variables with more meaningful names.

## Target release date 🗓️
10/16/2024

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/e4dd976d-0b7c-470c-a178-d08dcd14c3b3" /> | <video src="https://github.com/user-attachments/assets/74380f5d-715a-425f-8063-14f682093daf" /> |

## How to test 🧪
### Prerequisites
(How to setup test environment)
- Choose a Linode from your list of Linodes to work with and power it down.
- Visit the `Storage` tab for your chosen Linode.
- Resize any of the `Disks` in order to enable the `Add A Disk` option.
- Click on the `Add A Disk` button to open the `Create Disk` drawer.
- Click on the `Create from Image` radio option.

### Verification steps
(How to verify changes)
- Select an Image from the `Image` dropdown menu.
- Interact with any other field in the drawer (ex. `Label`, `Root Password`, `Size`).
- Verify that the value selected in the `Image` dropdown component remains after you have focused out of the component and focused into another field in the drawer.
- Verify that the `Add A Disk` feature works as expected.

## As an Author I have considered 🤔
*Check all that apply*
